runtime/api/src/main/java/org/qbicc/runtime/SerializedAsPrimitiveConstant.java

```diff
@@ -0,0 +1,22 @@
+package org.qbicc.runtime;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation to indicate that when an instance or static field is
+ * serialized to the qbicc initial heap that whatever value was stored
+ * in the field in the build time heap should be replaced by the
+ * zero-initializer appropriate to the type of the field in the runtime heap.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.FIELD)
+public @interface SerializedAsPrimitiveConstant {
+    /**
+     * The bits in the long will be cast to the appropriate primitive type
+     * and used as the replacement value for the field.
+     */
+    long bits();
```

Comment:
If this is the only property for the annotation, it should be called `Value` so you can have `@SerializedAsWhatever(10L)`.

Reply:
Went with a single property `value` of type `long` for now.